bundles/sirix-distributed/src/main/java/org/sirix/wikipedia/hadoop/RootElemAddition.java

```diff
@@ -0,0 +1,45 @@
+package org.sirix.wikipedia.hadoop;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
+/**
+ * Add a root-element to the sorted wikipedia-document.
+ * 
+ * @author Johannes Lichtenberger
+ *
+ */
+public final class RootElemAddition {
+
+	/** Wikipedia XML document. */
+	private static final File WIKIPEDIA_DOC = new File(new StringBuilder(
+			System.getProperty("user.home")).append(File.separator).append("Desktop")
+			.append(File.separator).append("wiki-articles-sorted.xml").toString());
+
+	/**
+	 * Main method.
+	 * 
+	 * @param args
+	 *          args[0]: input file
+	 * @throws IOException
+	 *           if an I/O error occured
+	 */
+	public static void main(final String[] args) throws IOException {
+		final FileWriter writer = new FileWriter(WIKIPEDIA_DOC);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *writer*. The resource is referenced in statements at the following lines: 31, 35, 37, 41, 42. The resource closure statement is at line: 43. There are other execution paths that do not contain closure statements, e.g., when exception is thrown by Writer.write. Either a) close *writer* in a try-finally block or b) close the resource by declaring *writer* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).